kfs-core/src/main/java/edu/arizona/kfs/pdp/batch/service/impl/PrepaidChecksServiceImpl.java

```diff
@@ -0,0 +1,507 @@
+        Batch batch = super.createNewBatch(prepaidChecksFile, super.getBaseFileName(incomingFileName));
+        businessObjectService.save(batch);
+
+        prepaidChecksFile.setBatchId(batch.getId());
+        status.setBatchId(batch.getId());
+        
+        // do warnings and set defaults
+        List<String> warnings = paymentFileValidationService.doSoftEdits(prepaidChecksFile);
+        status.setWarnings(warnings);
+        
+        // set needsTaxEmail to FALSE for all 
+        prepaidChecksFile.setTaxEmailRequired(false);
+        
+        // retrieve PaymentStatus BO for EXTR
+        PaymentStatus paymentStatus = (PaymentStatus) businessObjectService.findBySinglePrimaryKey(PaymentStatus.class, PdpConstants.PaymentStatusCodes.EXTRACTED);
+        
+        // retrieve DisbursementType BO for CHCK
+        DisbursementType disbursementType = businessObjectService.findBySinglePrimaryKey(DisbursementType.class, PdpConstants.DisbursementTypeCodes.CHECK);
+                
+        // store groups
+        for (PaymentGroup paymentGroup : prepaidChecksFile.getPaymentGroups()) {
+            // wire the batch in
+            paymentGroup.setBatchId(batch.getId());
+            paymentGroup.setBatch(batch);
+            
+            // refresh bank
+            paymentGroup.refreshReferenceObject(PdpPropertyConstants.CUSTOMER_BANK);
+            
+            // set paymentStatusCode to EXTR for all groups
+            paymentGroup.setPaymentStatusCode(PdpConstants.PaymentStatusCodes.EXTRACTED);            
+            paymentGroup.setPaymentStatus(paymentStatus);    
+                       
+            // force the disbursement type to CHCK
+            paymentGroup.setDisbursementTypeCode(PdpConstants.DisbursementTypeCodes.CHECK);
+            paymentGroup.setDisbursementType(disbursementType);
+            if (ObjectUtils.isNull(paymentGroup.getDisbursementType())) {
+                throw new RuntimeException("The paymentGroup's disbursementTypeCode ["+paymentGroup.getDisbursementTypeCode()+"] doesn't map " + 
+                        "to any DisbursementType object in the database.  This will cause the glPending Entries to be generated incorrectly, so " + 
+                        "we fail here.");
+            }
+            
+            businessObjectService.save(paymentGroup);
+
+            // Generate GL entries for the payments
+            glPendingTransactionService.generatePaymentGeneralLedgerPendingEntry(paymentGroup);
+        }
+
+        // send list of warnings
+        pdpEmailService.sendPrepaidChecksLoadEmail(prepaidChecksFile, warnings, incomingFileName);
+
+        LOG.debug("loadPrepaidChecks() was successful");
+        status.setLoadStatus(LoadPaymentStatus.LoadStatus.SUCCESS);
+    }
+        
+    @Override
+    protected PaymentFileLoad parsePaymentFile(BatchInputFileType paymentInputFileType, String incomingFileName, MessageMap errorMap) {
+    	PaymentFileLoad paymentFile = null;

```

Comment:
No change requested, since we are confined to the API here, but: paymentFile is being flagged as "always null" when making it inside of the catch blocks. Generally, such a state can be bug-prone, and ideally we'd want to separate into specifically disjoint methods to denote we know our filestream got hosed.

As it is, super has worked this way for who knows how long, it would be risky to refactor the foundation class, and this is probably not a hill to die on.

I did, however, want this noted should we ever need to look back on the PR, in the case of something pernicious occurring.